### PR TITLE
Expand Xiaomi Mijia MiBeacon support to e.g. MiFlora, Formaldehyde sensors

### PR DESCRIPTION
**Pull request by unrelentingtech**
- rename `temp_hum` to `sensor` 
- add support for skipping the capability byte that's in between the MAC and the data on some devices (e.g. MiFlora/VegTrug plant sensors)
- add more data types
- add support for multiple data items in one payload (not really tested as none of my devices seem to do this)

Roughly based on the best "documentation" for the protocol that is https://github.com/custom-components/ble_monitor.

### CODE REVIEW

File: main/broadcasters.c [hunk 1]
```diff
@@ -305,93 +305,166 @@ static broadcaster_ops_t eddystone_ops = {
...
-    }
-    else if (mijia_temp_hum->data_type == MIJIA_TEMP_HUM_DATA_TYPE_BATT)
-    {
-        sprintf(s, "%u", *mijia_temp_hum->data);
-        cb("BatteryLevel", s, ctx);
-    }
-    else if (mijia_temp_hum->data_type == MIJIA_TEMP_HUM_DATA_TYPE_TEMP_HUM)
+
+    if ((be16toh(mijia_header->frame_ctrl) & 0x4000) == 0)
+        return; /* No data */
```
Comment (shmuelzon) [inline main/broadcasters.c:387]: I, generally, prefer to avoid comments along side code, could you please move it above the condition as something like: `/* Check if any data is available */`

File: main/broadcasters.c [hunk 2]
```diff
@@ -305,93 +305,166 @@ static broadcaster_ops_t eddystone_ops = {
...
+
+    mijia_data_entry_t *mijia_data_entry = (mijia_data_entry_t *)(
+        (uint8_t *)mijia_header + sizeof(mijia_header_t) +
+        ((be16toh(mijia_header->frame_ctrl) & 0x2000) ? 1 : 0) /* skip capability byte */);
+
+    while ((uint8_t*)mijia_data_entry - (uint8_t*)mijia_header < len)
```
Comment (shmuelzon) [inline main/broadcasters.c:393]: `(uint8_t*)` ->`(uint8_t *)`

File: main/broadcasters.c [hunk 3]
```diff
@@ -305,93 +305,166 @@ static broadcaster_ops_t eddystone_ops = {
...
+        }
+        else if (mijia_data_entry->data_type == MIJIA_SENSOR_DATA_TYPE_FDH)
+        {
+            sprintf(s, "%.1f", le16toh(*(uint16_t *)mijia_data_entry->data) / 100.0);
+            cb("Formaldehyde", s, ctx);
+        }
+        else if (mijia_data_entry->data_type == MIJIA_SENSOR_DATA_TYPE_LUM)
+        {
+            uint32_t val = 0;
+            memcpy((uint8_t*)&val, mijia_data_entry->data, 3);
```
Comment (shmuelzon) [inline main/broadcasters.c:427]: Copying 3 bytes to a 4-byte type? This might cause endianness issues.
I think it would be safer to address the data as bytes a do something like
`uint32_t val = mijia_data_entry->data[0] | (mijia_data_entry->data[1] << 8) .....`.
Or maybe even
`sprintf(s, "%u", le32toh(*(uint32_t *)mijia_data_entry->data) & 0xff000000);`

File: main/broadcasters.c [hunk 4]
```diff
@@ -305,93 +305,166 @@ static broadcaster_ops_t eddystone_ops = {
...
+        {
+            sprintf(s, "%u", *mijia_data_entry->data);
+            cb("Switch", s, ctx);
+        }
+        else if (mijia_data_entry->data_type == MIJIA_SENSOR_DATA_TYPE_CONSUM)
+        {
+            sprintf(s, "%u", *mijia_data_entry->data);
+            cb("Consumable", s, ctx);
+        }
+        mijia_data_entry = (mijia_data_entry_t*)(
```
Comment (shmuelzon) [inline main/broadcasters.c:460]: `(mijia_data_entry_t*)` -> `(mijia_data_entry_t *)`

File: main/broadcasters.c [hunk 5]
```diff
@@ -305,93 +305,166 @@ static broadcaster_ops_t eddystone_ops = {
...
+            (uint8_t*)mijia_data_entry + 3 + mijia_data_entry->data_len);
```
Comment (shmuelzon) [inline main/broadcasters.c:461]: `(uint8_t*)` -> `(uint8_t *)`

**Review by shmuelzon — CHANGES_REQUESTED** [review]
Thank you so much for this pull request, I really appreciate it.
I do have one comment/suggestion, the rest are just conventions/nitpicking.

**Comment by unrelentingtech** [comment]
Updated with fixes and rebased

**Review by shmuelzon — APPROVED** [review]

**Comment by shmuelzon** [comment]
Thanks!
